pkg/metrics/bpf.go

```diff
@@ -36,55 +56,89 @@ func newbpfCollector() *bpfCollector {
 }
 
 func (s *bpfCollector) Describe(ch chan<- *prometheus.Desc) {
-	ch <- s.bpfMapsMemory
-	ch <- s.bpfProgMemory
+	prometheus.DescribeByCollect(s, ch)
 }
 
 type memoryEntry struct {
 	BytesMemlock uint64 `json:"bytes_memlock"`
 }
 
-func getMemoryUsage(typ string) (uint64, error) {
+func getBPFUsage(typ string) (bpfUsage, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 	cmd := exec.CommandContext(ctx, "bpftool", "-j", typ, "show")
 	out, err := cmd.Output()
 	if err != nil {
-		return 0, fmt.Errorf("unable to get bpftool output: %w", err)
+		return bpfUsage{}, fmt.Errorf("unable to get bpftool output: %w", err)
 	}
 
 	var memoryEntries []memoryEntry
 	err = json.Unmarshal(out, &memoryEntries)
 	if err != nil {
-		return 0, fmt.Errorf("unable to unmarshal bpftool output: %w", err)
+		return bpfUsage{}, fmt.Errorf("unable to unmarshal bpftool output: %w", err)
 	}
 	var totalMem uint64
 	for _, entry := range memoryEntries {
 		totalMem += entry.BytesMemlock
 	}
-	return totalMem, nil
+
+	return bpfUsage{
```

Comment:
Your changes are consistent with how things where done before, but I find it a bit unexpected that we don't only count maps prefixed with `cilium_`

Reply:
do you feel we should we change it as part of this PR or through a separate one though?